fastlane/Fastfile

```diff
@@ -30,27 +30,29 @@ end
 
 desc "Release a new version"
 lane :release do |options|
-  ensure_git_branch(branch: 'main') # We can only release on default branch
   ensure_git_status_clean unless options[:no_ensure_clean]
 
+  # Ensure we have a release type
+  UI.user_error!("Please use type parameter with one of the options: type:patch, type:minor, type:major") unless ["patch", "minor", "major"].include?(options[:type])
+
+  # Ensure We Have the GITHUB Token
   UI.user_error!("Please set GITHUB_TOKEN environment value. See https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token") unless !ENV["GITHUB_TOKEN"].nil?
 
-  if (!options[:version].nil?) # User passed a version, use it
-    version_number = options.fetch(:version)
-    increment_version_number_in_plist(version_number: version_number, xcodeproj: "StreamChat.xcodeproj", target: "StreamChat")
-  else
-    UI.user_error!("Please use type parameter with one of the options: type:patch, type:minor, type:major") unless ["patch", "minor", "major"].include?(options[:type])
-    version_number = increment_version_number_in_plist(bump_type: options[:type], xcodeproj: "StreamChat.xcodeproj", target: "StreamChat")
-  end
+  # Version Number
+  version_number = increment_version_number_in_plist(bump_type: options[:type], xcodeproj: "StreamChat.xcodeproj", target: "StreamChat")
 
-  set_SDK_version(version: version_number)
-
-  increment_version_number_in_plist(version_number: version_number, xcodeproj: "StreamChat.xcodeproj", target: "StreamChatUI")
+  # Checkout New Release Branch
+  sh("git checkout -b " + "release/" + version_number)
 
   if git_tag_exists(tag: version_number)
     UI.user_error!("Tag for version #{version_number} already exists!")
   end
```

Comment:
We should move this check before creation of the branch

Reply:
Agree